関数をファイルに分ける作業（名前がよくわからない）

844196 opened the pull request:
- [x] `usage()`
- [x] `error()`
- [x] `createRandomNumber()`
- [x] オプション解析部
- [x] バージョン出力部

844196 committed:
renge.srcをbin/へ移動

bin/をgit管理対象へ

844196 commented:
よく見たら最新が2.1じゃないか（呆れ）

sasairc commented:
@844196 
分割したら`Makefile`等書いてPull Requestを送りますので、よろしくお願いします。

844196 commented:
@sasairc :oden:

844196 committed:
バージョン情報をversion()へ分離

844196 committed:
usage()を分離

844196 committed:
error()を分離

844196 committed:
オプションパーサーをoption_parse.bashへ分離

844196 committed:
createRandomNumberを分離

844196 committed:
ファイル移動

844196 commented:
だいたい分割完了

```
.
├── bin
│   └── renge.src
├── etc
│   ├── _renge.zsh
│   └── old-renge-quotes
├── lib
│   └── renge
│       ├── createRandomNumber.bash
│       ├── error.bash
│       ├── option_parse.bash
│       ├── usage.bash
│       └── version.bash
├── share
│   └── renge
│       └── renge-quotes
├── test
│   └── test.sh
├── .travis.yml
├── LICENSE
├── Makefile
└── README.md
```

844196 commented:
テストコード

``` bash
cat ./bin/renge.src | sed -e "s/PREPRE/\./g" -e "s/_LIBDIR/\.\/lib\/renge/g" | bash             # 普通の
cat ./bin/renge.src | sed -e "s/PREPRE/\./g" -e "s/_LIBDIR/\.\/lib\/renge/g" | bash -s -- -v    # バージョン出力
cat ./bin/renge.src | sed -e "s/PREPRE/\./g" -e "s/_LIBDIR/\.\/lib\/renge/g" | bash -s -- -h    # ヘルプ出力
cat ./bin/renge.src | sed -e "s/PREPRE/\./g" -e "s/_LIBDIR/\.\/lib\/renge/g" | bash -s -- -l    # 全行出力
cat ./bin/renge.src | sed -e "s/PREPRE/\./g" -e "s/_LIBDIR/\.\/lib\/renge/g" | bash -s -- -n 84 # 行番号指定出力
```

``` shellsession
$ time bash -c 'for i in {1..100};{ cat ./bin/renge.src | sed -e "s/PREPRE/\./g" -e "s/_LIBDIR/\.\/lib\/renge/g" | bash >/dev/null;}'
real    0m1.409s
user    0m0.407s
sys 0m0.087s
```

844196 commented:
![b71c1da6](https://cloud.githubusercontent.com/assets/4990822/9848493/9647c56c-5b1c-11e5-841b-78994db51340.jpg)

@sasairc というわけでおねがいします。PRの宛先は`w_2.1`でおねがいします。

sasairc commented:
@844196 
`etc/`の下にある`old-renge-quotes`は如何なさいましょう
`make install`時に、`$(PREFIX)/share/renge`下に配置しますかね？

844196 commented:
@sasairc 

> make install時に、$(PREFIX)/share/renge下に配置しますかね？

本当は清く消したかったんですが、名残惜しいので置いてあるだけです。
特にコピー処理はいらないです。

sasairc commented:
@844196 
おかのした
`_renge.zsh`はどうしましょう。
インストールするのであれば、ソースツリー内も`share/renge/compdef/_renge.zsh`とした方が良さ気な気がします。

844196 commented:
@sasairc 
おかのした

**追記**
これからコミット履歴を改変します
フォーク済みの場合は`HEAD`が変わるので注意してください

844196 committed:
ファイル移動

deleted:    .gitignore
deleted:    old-renge-quotes
renamed:    _renge.zsh -> share/renge/compdef/_renge.zsh
renamed:    renge-quotes -> share/renge/renge-quotes

844196 commented:
新しいディレクトリ配置

```
.
|-- bin
|   `-- renge.src
|-- lib
|   `-- renge
|       |-- createRandomNumber.bash
|       |-- error.bash
|       |-- option_parse.bash
|       |-- usage.bash
|       `-- version.bash
|-- share
|   `-- renge
|       |-- compdef
|       |   `-- _renge.zsh
|       `-- renge-quotes
|-- test
|   `-- test.sh
|-- .travis.yml
|-- LICENSE
|-- Makefile
`-- README.md
```

844196 commented:
@sasairc 
改変完了しました。@05a2c8f が`HEAD`になります

sasairc committed:
トップディレクトリにMakefileを配置

sasairc committed:
サブディレクトリにMakefileを配置

sasairc committed:
renamed:    etc/_renge.zsh -> share/renge/compdef/_renge.zsh

sasairc committed:
Merge remote-tracking branch 'upstream/w_2.1' into w_2.1

Conflicts:
	etc/old-renge-quotes

sasairc commented:
@844196 
#25 で`w_2.1`宛に送りました、ご確認下さい

844196 committed:
Merge pull request #25 from sasairc/w_2.1

Makefile等の配置

844196 commented:
@sasairc 
テスト通過しました。ありがとうございます。

`w_2.1`は単体テスト追加後`master`へマージします。

sasairc commented:
@844196 
いえいえ、こちらこそありがとうございます。
次のリリースを楽しみにしています。

844196 commented:
げんきなうちにまーじ

844196 committed:
旧辞書ファイルを削除

844196 committed:
ver 2.2
